frontend/store/items.ts

```diff
@@ -1,193 +1,174 @@
+    },
+    /**
+     * Deletes a single or multiple items from the store
+     *
+     * @param payload
+     */
+    delete(payload: string | string[]): void {
+      if (!Array.isArray(payload)) {
+        payload = [payload];
+      }
 
-    newAllIds = union(newAllIds, Object.keys(items));
-    state.allIds = Object.freeze(newAllIds);
-  },
-  DELETE_ITEM(state: ItemsState, { id }: { id: string }) {
-    delete state.byId[id];
+      for (const id of payload) {
+        Vue.delete(this.byId, id);
+      }
+    },
+    /**
+     * Associate an item that has children with its children
+     *
+     * @param parent
+     * @param children
+     * @returns - The children of the item
+     */
+    addCollection(parent: BaseItemDto, children: BaseItemDto[]): BaseItemDto[] {
+      if (!parent.Id) {
+        throw new Error("Parent item doesn't have an Id");
+      }
 
-    const idx = state.allIds.indexOf(id);
+      const childIds = [];
 
-    if (idx > -1) {
-      const newAllIds = Array.from(state.allIds);
+      for (const child of children) {
+        if (child.Id) {
+          if (!this.getItemById(child.Id)) {
+            this.add(child);
+          }
 
-      newAllIds.splice(idx, 1);
-      state.allIds = Object.freeze(newAllIds);
-    }
-  },
-  ADD_BOXSET_CHILDREN(state: ItemsState, { childrenList, itemId }) {
-    if (!itemId) {
-      throw new Error('itemId is undefined');
-    }
+          childIds.push(child.Id);
+        }
+      }
 
-    if (!childrenList) {
-      return [];
-    }
-    //  Error('childrenList is undefined');
+      Vue.set(this.collectionById, parent.Id, childIds);
+
+      return this.getChildrenOfParent(parent.Id) as BaseItemDto[];
+    },
+    /**
+     * Fetches a parent and its children and adds thecollection to the store
+     *
+     * @param parentId
+     */
+    async fetchAndAddCollection(
+      parentId: string | undefined
+    ): Promise<BaseItemDto[] | undefined> {
+      const auth = authStore();
+
+      if (parentId && !this.getItemById(parentId)) {
+        const parentItem = (
+          await this.$nuxt.$api.items.getItems({
+            userId: auth.currentUserId,
+            ids: [parentId],
+            fields: allFields
+          })
+        ).data;
+
+        if (!parentItem.Items?.[0]) {
+          throw new Error("This parent doesn't exist");
+        }
 
-    const children: { [key: string]: BaseItemDto[] } = {};
+        this.add(parentItem.Items[0]);
+      }
 
-    childrenList.forEach((item: BaseItemDto) => {
-      if (item.Type && typeof item.Type === 'string') {
-        if (!children[item.Type]) {
-          children[item.Type] = [];
-        }
+      const childItems = (
+        await this.$nuxt.$api.items.getItems({
+          userId: auth.currentUserId,
+          parentId,
+          fields: allFields
+        })
+      ).data;
 
-        children[item.Type].push(item);
+      if (childItems.Items) {
+        return this.add(childItems.Items) as BaseItemDto[];
```

Comment:
You created the method above but didn't use it later:
```suggestion
        const parent = this.getItemById(parentId);
      if (childItems.Items && parent) {
        return this.addCollection(parent, childItems.Items) as BaseItemDto[];
      }
```

Reply:
Good catch!